### PR TITLE
8324681: Replace NULL with nullptr in HotSpot jtreg test native code files

### PR DESCRIPTION
**Pull request by coleenp**
This mechanically replaces NULL with nullptr in hpp/cpp native files in test native code.  This didn't attempt to change NULL in comments to say null because nullptr is generally the right thing for the comment to say.  It does attempt to change NULL to "null" rather than "nullptr" in strings.  Any changes for "nullptr" to "null" in comments can be changed in a future RFE in a smaller patch. I didn't see any when it was scrolling by to make my script more complicated.

Ran tier1-4 testing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324681](https://bugs.openjdk.org/browse/JDK-8324681): Replace NULL with nullptr in HotSpot jtreg test native code files (**Enhancement** - P4)

### Reviewers
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - Committer) ⚠️ Review applies to [33786c7d](https://git.openjdk.org/jdk/pull/17593/files/33786c7dd208b19b5cc23295cd41dc69a03529f6)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [6eb051ed](https://git.openjdk.org/jdk/pull/17593/files/6eb051edf88efc5b74220b98dbc987629fdac075)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [6eb051ed](https://git.openjdk.org/jdk/pull/17593/files/6eb051edf88efc5b74220b98dbc987629fdac075)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17593/head:pull/17593` \
`$ git checkout pull/17593`

Update a local copy of the PR: \
`$ git checkout pull/17593` \
`$ git pull https://git.openjdk.org/jdk.git pull/17593/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17593`

View PR using the GUI difftool: \
`$ git pr show -t 17593`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17593.diff">https://git.openjdk.org/jdk/pull/17593.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17593#issuecomment-1912633444)

### CODE REVIEW

File: test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadTest/libVThreadTest.cpp [hunk 1]
```diff
@@ -423,18 +423,18 @@ test_GetLocal(jvmtiEnv *jvmti, JNIEnv *jni, jthread cthread, jthread vthread, co
                " to return JVMTI_ERROR_INVALID_SLOT or JVMTI_ERROR_TYPE_MISMATCH");
   }
 
-  // #5: Test JVMTI GetLocalObject function with NULL value_ptr
-  err = jvmti->GetLocalObject(vthread, depth, SlotObj, NULL);
+  // #5: Test JVMTI GetLocalObject function with nullptr value_ptr
```
Comment (kevinjwalls) [inline test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadTest/libVThreadTest.cpp:426]: "with null value_ptr" ?

**Review by kevinjwalls — APPROVED** [review]
That is long.  Hypnotic to read. One very trivial nit, use or ignore as required, but all good that I can see.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks for doing the text changes as well, they are a necessary part of the cleanup.

A number of files are missing copyright updates - the ones I spotted all had a single copyright year so maybe your script missed them?

My browser found 21 places where nullptr is cast to something else, which should no longer be needed.

Thanks.

**Comment by coleenp** [comment]
Thanks Kevin, Kim and David for wading through this.  If there are other changes we can address them separately preserving your eyeballs.  My copyright script was broken so I fixed it.  I'll wait for GHA to make sure I didn't break anything before integrating.

**Comment by coleenp** [comment]
macos-aarch64 build failure in GHA appears unrelated, internal testing passed.
/integrate

**Pull request closed without merging**